pkg/apps/apis/apps/v1/types.go

```diff
@@ -49,10 +50,12 @@ type DeploymentConfigSpec struct {
 	// Triggers determine how updates to a DeploymentConfig result in new deployments. If no triggers
 	// are defined, a new deployment can only occur as a result of an explicit client update to the
 	// DeploymentConfig with a new LatestVersion. If null, defaults to having a config change trigger.
+	// +optional
 	Triggers DeploymentTriggerPolicies `json:"triggers" protobuf:"bytes,2,rep,name=triggers"`
 
 	// Replicas is the number of desired replicas.
-	Replicas int32 `json:"replicas" protobuf:"varint,3,opt,name=replicas"`
+	// +optional
+	Replicas *int32 `json:"replicas" protobuf:"varint,3,opt,name=replicas"`
```

Comment:
@smarterclayton how is this not a breaking change?

Reply:
The value still defaults.  No client ever sees this being null.  You can make a required field not required as long as you provide a default.

Reply:
It will break everyone using our Go client :/

But I guess on the pure API front the serialization will go fine when we default unless client is able to see the object before defaulting happens (say with GetOptions.IncludeUninitialized) but I think we initialize before storing the object.

Reply:
> it will break everyone using our Go client :/

we don't have a go client yet.

> But I guess on the pure API front the serialization will go fine when we default 

yes

> unless client is able to see the object before defaulting happens

it cannot

Reply:
> we don't have a supported go client yet.

so we don't "support" client/clientsets in origin codebase? because people use that...

Reply:
We don't guarantee they don't change.  Users have to track updates to code.